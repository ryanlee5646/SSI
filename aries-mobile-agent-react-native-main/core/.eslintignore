node_modules
__tests__
ios
android

# generated by bob
lib/
bifold-app/
generate-app-package-json.js