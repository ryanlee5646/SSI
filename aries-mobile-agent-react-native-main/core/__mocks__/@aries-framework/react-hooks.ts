import { CredentialRecord, ProofRecord } from '@aries-framework/core'

const useCredentials = jest.fn().mockReturnValue({ credentials: [] } as any)
const useCredentialByState = jest.fn().mockReturnValue([] as CredentialRecord[])
const useProofByState = jest.fn().mockReturnValue([] as ProofRecord[])
const mockCredentialModule = {
  acceptOffer: jest.fn(),
  declineOffer: jest.fn(),
}
const mockProofModule = {
  getRequestedCredentialsForProofRequest: jest.fn(),
  acceptRequest: jest.fn(),
  declineRequest: jest.fn(),
}
const useAgent = () => ({
  agent: {
    credentials: mockCredentialModule,
    proofs: mockProofModule,
  },
})
const useCredentialById = jest.fn()
const useProofById = jest.fn()
const useConnectionById = jest.fn()

export {
  useAgent,
  useConnectionById,
  useCredentials,
  useCredentialById,
  useCredentialByState,
  useProofById,
  useProofByState,
}
