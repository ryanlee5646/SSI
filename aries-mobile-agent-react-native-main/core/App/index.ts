/* eslint-disable import/no-cycle */
import type { OnboardingStyleSheet } from './screens/Onboarding'

import { Agent } from '@aries-framework/core'
import AgentProvider from '@aries-framework/react-hooks'

import indyLedgers from '../configs/ledgers/indy'

import Button, { ButtonType } from './components/buttons/Button'
import CheckBoxRow from './components/inputs/CheckBoxRow'
import ErrorModal from './components/modals/ErrorModal'
import InfoTextBox from './components/texts/InfoTextBox'
import toastConfig from './components/toast/ToastConfig'
import RootStack from './navigators/RootStack'
import OnboardingPages from './screens/OnboardingPages'
import Splash from './screens/Splash'
import Terms from './screens/Terms'

export { LocalStorageKeys } from './constants'
export { initLanguages, initStoredLanguage, translationResources } from './localization'
export { ConfigurationProvider, useConfiguration } from './contexts/configuration'
export { StoreProvider, StoreContext } from './contexts/store'
export { DispatchAction } from './contexts/reducers/store'
export { ThemeProvider, useTheme } from './contexts/theme'
export { ColorPallet } from './theme'
export { theme } from './theme'
export { testIdWithKey } from './utils/testable'
export { Screens } from './types/navigators'
export { createStyles } from './screens/OnboardingPages'

export type { StoreProviderProps } from './contexts/store'
export type { Theme } from './theme'
export type { ConfigurationContext } from './contexts/configuration'
export type { Onboarding as OnboardingState } from './types/state'
export type { GenericFn } from './types/fn'
export type { AuthenticateStackParams } from './types/navigators'
export type { OnboardingStyleSheet }

export {
  indyLedgers,
  Agent,
  AgentProvider,
  Button,
  ButtonType,
  CheckBoxRow,
  InfoTextBox,
  ErrorModal,
  toastConfig,
  RootStack,
  OnboardingPages,
  Splash,
  Terms,
}
