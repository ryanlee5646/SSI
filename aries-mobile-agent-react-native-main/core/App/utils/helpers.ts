import {
  ConnectionRecord,
  CredentialMetadataKeys,
  CredentialRecord,
  ProofRecord,
  RequestedAttribute,
} from '@aries-framework/core'
import { useConnectionById, useCredentialById, useProofById } from '@aries-framework/react-hooks'
import { parseUrl } from 'query-string'

import { Attribute } from '../types/record'

export function parseSchema(schemaId?: string): { name: string; version: string } {
  let name = 'Credential'
  let version = ''
  if (schemaId) {
    const schemaIdRegex = /(.*?):([0-9]):([a-zA-Z .\-_0-9]+):([a-z0-9._-]+)$/
    const schemaIdParts = schemaId.match(schemaIdRegex)
    if (schemaIdParts?.length === 5) {
      name = `${schemaIdParts?.[3].replace(/_|-/g, ' ')}`
        .split(' ')
        .map((schemaIdPart) => schemaIdPart.charAt(0).toUpperCase() + schemaIdPart.substring(1))
        .join(' ')
      version = schemaIdParts?.[4]
    }
  }
  return { name, version }
}

export function credentialSchema(credential: CredentialRecord): string | undefined {
  return credential.metadata.get(CredentialMetadataKeys.IndyCredential)?.schemaId
}

export function parsedSchema(credential: CredentialRecord): { name: string; version: string } {
  return parseSchema(credentialSchema(credential))
}

/**
 * Adapted from https://stackoverflow.com/questions/3426404/create-a-hexadecimal-colour-based-on-a-string-with-javascript
 */
export function hashCode(s: string): number {
  return s.split('').reduce((hash, char) => char.charCodeAt(0) + ((hash << 5) - hash), 0)
}

export function hashToRGBA(i: number) {
  const colour = (i & 0x00ffffff).toString(16).toUpperCase()
  return '#' + '00000'.substring(0, 6 - colour.length) + colour
}

export function credentialRecordFromId(credentialId?: string): CredentialRecord | void {
  if (credentialId) {
    return useCredentialById(credentialId)
  }
}

export function connectionRecordFromId(connectionId?: string): ConnectionRecord | void {
  if (connectionId) {
    return useConnectionById(connectionId)
  }
}

export function proofRecordFromId(proofId?: string): ProofRecord | void {
  if (proofId) {
    return useProofById(proofId)
  }
}

export function getConnectionName(connection: ConnectionRecord | void): string | void {
  if (!connection) {
    return
  }
  return connection?.alias || connection?.invitation?.label
}

export function firstAttributeCredential(attributes: RequestedAttribute[], revoked = true): RequestedAttribute | null {
  if (!attributes.length) {
    return null
  }

  let first = null
  const firstNonRevoked = attributes.filter((attribute) => !attribute.revoked)[0]
  if (firstNonRevoked) {
    first = firstNonRevoked
  } else if (attributes.length && revoked) {
    first = attributes[0]
  }

  if (!first?.credentialInfo) {
    return null
  }

  return first
}

export const isRedirection = (url: string): boolean => {
  const queryParams = parseUrl(url).query
  return !(queryParams['c_i'] || queryParams['d_m'])
}

export const processProofAttributes = (
  proof: ProofRecord,
  attributes: Record<string, RequestedAttribute[]> = {}
): Attribute[] => {
  const processedAttributes = [] as Attribute[]
  const { requestedAttributes: requestedProofAttributes } = proof.requestMessage?.indyProofRequest || {}

  requestedProofAttributes?.forEach(({ name, names }, attributeName) => {
    const firstCredential = firstAttributeCredential(attributes[attributeName] || [])
    const credentialAttributes = names?.length ? names : [name || attributeName]
    credentialAttributes.forEach((attribute) => {
      processedAttributes.push({
        name: attribute,
        value: firstCredential?.credentialInfo?.attributes[attribute] || null,
        revoked: firstCredential?.revoked || false,
        credentialId: firstCredential?.credentialId,
      })
    })
  })
  return processedAttributes
}
