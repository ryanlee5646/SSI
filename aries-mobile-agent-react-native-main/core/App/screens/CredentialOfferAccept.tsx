import { CredentialState } from '@aries-framework/core'
import { useCredentialById } from '@aries-framework/react-hooks'
import { useNavigation } from '@react-navigation/core'
import React, { useEffect, useState } from 'react'
import { useTranslation } from 'react-i18next'
import { Modal, StyleSheet, Text, View } from 'react-native'
import { SafeAreaView } from 'react-native-safe-area-context'

import CredentialAdded from '../components/animated/CredentialAdded'
import CredentialPending from '../components/animated/CredentialPending'
import Button, { ButtonType } from '../components/buttons/Button'
import { useTheme } from '../contexts/theme'
import { Screens, TabStacks } from '../types/navigators'
import { testIdWithKey } from '../utils/testable'

const connectionTimerDelay = 5000 // in ms

enum DeliveryStatus {
  Pending,
  Completed,
  Declined,
}

export interface CredentialOfferAcceptProps {
  visible: boolean
  credentialId: string
}

const CredentialOfferAccept: React.FC<CredentialOfferAcceptProps> = ({ visible, credentialId }) => {
  const { t } = useTranslation()
  const [shouldShowDelayMessage, setShouldShowDelayMessage] = useState<boolean>(false)
  const [credentialDeliveryStatus, setCredentialDeliveryStatus] = useState<DeliveryStatus>(DeliveryStatus.Pending)
  const [timerDidFire, setTimerDidFire] = useState<boolean>(false)
  const [timer, setTimer] = useState<NodeJS.Timeout>()
  const credential = useCredentialById(credentialId)
  const navigation = useNavigation()
  const { ColorPallet, TextTheme } = useTheme()
  const styles = StyleSheet.create({
    container: {
      flexGrow: 1,
      flexDirection: 'column',
      backgroundColor: ColorPallet.brand.primaryBackground,
      paddingHorizontal: 25,
      paddingTop: 20,
    },
    image: {
      marginTop: 20,
    },
    messageContainer: {
      alignItems: 'center',
    },
    messageText: {
      fontWeight: 'normal',
      textAlign: 'center',
      marginTop: 90,
    },
    controlsContainer: {
      marginTop: 'auto',
      marginBottom: 20,
    },
    delayMessageText: {
      textAlign: 'center',
      marginTop: 20,
    },
  })

  if (!credential) {
    throw new Error('Unable to fetch credential from AFJ')
  }

  const onBackToHomeTouched = () => {
    navigation.getParent()?.navigate(TabStacks.HomeStack, { screen: Screens.Home })
  }

  const onDoneTouched = () => {
    navigation.getParent()?.navigate(TabStacks.CredentialStack, { screen: Screens.Credentials })
  }

  useEffect(() => {
    if (credential.state === CredentialState.CredentialReceived || credential.state === CredentialState.Done) {
      timer && clearTimeout(timer)
      setCredentialDeliveryStatus(DeliveryStatus.Completed)
    }
  }, [credential])

  useEffect(() => {
    if ((timerDidFire || credentialDeliveryStatus !== DeliveryStatus.Pending) && !visible) {
      return
    }

    const timer = setTimeout(() => {
      setShouldShowDelayMessage(true)
      setTimerDidFire(true)
    }, connectionTimerDelay)

    setTimer(timer)

    return () => {
      timer && clearTimeout(timer)
    }
  }, [visible])

  return (
    <Modal visible={visible} transparent={true} animationType={'none'}>
      <SafeAreaView style={[styles.container]}>
        <View style={[styles.messageContainer]}>
          {credentialDeliveryStatus === DeliveryStatus.Pending && (
            <Text style={[TextTheme.headingThree, styles.messageText]} testID={testIdWithKey('CredentialOnTheWay')}>
              {t('CredentialOffer.CredentialOnTheWay')}
            </Text>
          )}

          {credentialDeliveryStatus === DeliveryStatus.Completed && (
            <Text
              style={[TextTheme.headingThree, styles.messageText]}
              testID={testIdWithKey('CredentialAddedToYourWallet')}
            >
              {t('CredentialOffer.CredentialAddedToYourWallet')}
            </Text>
          )}
        </View>

        <View style={[styles.image, { minHeight: 250, alignItems: 'center', justifyContent: 'flex-end' }]}>
          {credentialDeliveryStatus === DeliveryStatus.Completed && <CredentialAdded />}
          {credentialDeliveryStatus === DeliveryStatus.Pending && <CredentialPending />}
        </View>

        {shouldShowDelayMessage && credentialDeliveryStatus === DeliveryStatus.Pending && (
          <Text style={[TextTheme.normal, styles.delayMessageText]} testID={testIdWithKey('TakingTooLong')}>
            {t('Connection.TakingTooLong')}
          </Text>
        )}

        <View style={[styles.controlsContainer]}>
          {shouldShowDelayMessage && credentialDeliveryStatus === DeliveryStatus.Pending && (
            <Button
              title={t('Loading.BackToHome')}
              accessibilityLabel={t('Loading.BackToHome')}
              testID={testIdWithKey('BackToHome')}
              onPress={onBackToHomeTouched}
              buttonType={ButtonType.Secondary}
            />
          )}

          {credentialDeliveryStatus === DeliveryStatus.Completed && (
            <Button
              title={t('Global.Done')}
              accessibilityLabel={t('Global.Done')}
              testID={testIdWithKey('Done')}
              onPress={onDoneTouched}
              buttonType={ButtonType.Primary}
            />
          )}
        </View>
      </SafeAreaView>
    </Modal>
  )
}

export default CredentialOfferAccept
