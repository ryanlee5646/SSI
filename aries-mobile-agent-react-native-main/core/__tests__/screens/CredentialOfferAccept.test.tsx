import { CredentialRecord, CredentialState } from '@aries-framework/core'
import { useCredentialById } from '@aries-framework/react-hooks'
import { render, waitFor } from '@testing-library/react-native'
import fs from 'fs'
import path from 'path'
import React from 'react'

import CredentialOfferAccept from '../../App/screens/CredentialOfferAccept'
import { testIdWithKey } from '../../App/utils/testable'
import timeTravel from '../util/timetravel'

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper')

const credentialId = '123'
const credentialPath = path.join(__dirname, '../fixtures/degree-credential.json')
const credential = JSON.parse(fs.readFileSync(credentialPath, 'utf8'))
const credentialRecord = new CredentialRecord(credential)
// TODO:(jl) Make a fn to revive JSON dates properly and pass to `parse`
credentialRecord.createdAt = new Date(credentialRecord.createdAt)

// @ts-ignore
useCredentialById.mockReturnValue(credentialRecord)

describe('displays a credential accept screen', () => {
  test('renders correctly', () => {
    const tree = render(<CredentialOfferAccept visible={true} credentialId={credentialId} />)

    const doneButton = tree.queryByTestId('Done')
    const backToHomeButton = tree.queryByTestId(testIdWithKey('BackToHome'))

    expect(tree).toMatchSnapshot()
    expect(doneButton).toBeNull()
    expect(backToHomeButton).toBeNull()
    expect(useCredentialById).toBeCalledWith(credentialId)
  })

  test('transitions to taking too delay message', async () => {
    const tree = render(<CredentialOfferAccept visible={true} credentialId={credentialId} />)

    await waitFor(() => {
      timeTravel(10000)
    })

    const backToHomeButton = tree.getByTestId(testIdWithKey('BackToHome'))
    const doneButton = tree.queryByTestId(testIdWithKey('Done'))

    expect(tree).toMatchSnapshot()
    expect(backToHomeButton).not.toBeNull()
    expect(doneButton).toBeNull()
  })

  test('transitions to offer accepted', () => {
    credentialRecord.state = CredentialState.CredentialReceived

    const tree = render(<CredentialOfferAccept visible={true} credentialId={credentialId} />)

    const doneButton = tree.getByTestId(testIdWithKey('Done'))
    const backToHomeButton = tree.queryByTestId(testIdWithKey('BackToHome'))

    expect(tree).toMatchSnapshot()
    expect(doneButton).not.toBeNull()
    expect(backToHomeButton).toBeNull()
  })
})
