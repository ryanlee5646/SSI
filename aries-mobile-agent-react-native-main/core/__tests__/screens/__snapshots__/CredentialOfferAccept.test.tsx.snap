// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays a credential accept screen renders correctly 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <RNCSafeAreaView
    style={
      Array [
        Object {
          "backgroundColor": "#000000",
          "flexDirection": "column",
          "flexGrow": 1,
          "paddingHorizontal": 25,
          "paddingTop": 20,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 26,
              "fontWeight": "bold",
            },
            Object {
              "fontWeight": "normal",
              "marginTop": 90,
              "textAlign": "center",
            },
          ]
        }
        testID="com.ariesbifold:id/CredentialOnTheWay"
      >
        CredentialOffer.CredentialOnTheWay
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "marginTop": 20,
          },
          Object {
            "alignItems": "center",
            "justifyContent": "flex-end",
            "minHeight": 250,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "flexDirection": "column",
            },
          ]
        }
      >
        <
          height={110}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "marginTop": -30,
                "position": "absolute",
              },
            ]
          }
          width={110}
        />
        <View
          style={
            Object {
              "opacity": 0,
              "transform": Array [
                Object {
                  "translateY": -90,
                },
              ],
            }
          }
        >
          <
            height={110}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "marginLeft": 10,
                  "position": "absolute",
                },
              ]
            }
            width={110}
          />
        </View>
        <
          height={140}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
            ]
          }
          width={140}
        />
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "marginBottom": 20,
            "marginTop": "auto",
          },
        ]
      }
    />
  </RNCSafeAreaView>
</Modal>
`;

exports[`displays a credential accept screen transitions to offer accepted 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <RNCSafeAreaView
    style={
      Array [
        Object {
          "backgroundColor": "#000000",
          "flexDirection": "column",
          "flexGrow": 1,
          "paddingHorizontal": 25,
          "paddingTop": 20,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 26,
              "fontWeight": "bold",
            },
            Object {
              "fontWeight": "normal",
              "marginTop": 90,
              "textAlign": "center",
            },
          ]
        }
        testID="com.ariesbifold:id/CredentialAddedToYourWallet"
      >
        CredentialOffer.CredentialAddedToYourWallet
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "marginTop": 20,
          },
          Object {
            "alignItems": "center",
            "justifyContent": "flex-end",
            "minHeight": 250,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "flexDirection": "column",
            },
          ]
        }
      >
        <View
          collapsable={false}
          nativeID="animatedComponent"
          style={
            Object {
              "alignItems": "center",
              "marginBottom": 10,
              "opacity": 0,
            }
          }
        >
          <
            height={45}
            width={45}
          />
        </View>
        <View>
          <
            height={110}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "marginTop": -30,
                  "position": "absolute",
                },
              ]
            }
            width={110}
          />
          <View
            collapsable={false}
            nativeID="animatedComponent"
            style={
              Object {
                "opacity": 0,
                "transform": Array [
                  Object {
                    "translateY": -90,
                  },
                ],
              }
            }
          >
            <
              height={110}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "marginLeft": 10,
                    "position": "absolute",
                  },
                ]
              }
              width={110}
            />
          </View>
          <
            height={140}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
              ]
            }
            width={140}
          />
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "marginBottom": 20,
            "marginTop": "auto",
          },
        ]
      }
    >
      <View
        accessibilityLabel="Global.Done"
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#42803E",
            "borderRadius": 4,
            "opacity": 1,
            "padding": 16,
          }
        }
        testID="com.ariesbifold:id/Done"
      >
        <Text
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 18,
                "fontWeight": "bold",
                "textAlign": "center",
              },
              false,
            ]
          }
        >
          Global.Done
        </Text>
      </View>
    </View>
  </RNCSafeAreaView>
</Modal>
`;

exports[`displays a credential accept screen transitions to taking too delay message 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <RNCSafeAreaView
    style={
      Array [
        Object {
          "backgroundColor": "#000000",
          "flexDirection": "column",
          "flexGrow": 1,
          "paddingHorizontal": 25,
          "paddingTop": 20,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 26,
              "fontWeight": "bold",
            },
            Object {
              "fontWeight": "normal",
              "marginTop": 90,
              "textAlign": "center",
            },
          ]
        }
        testID="com.ariesbifold:id/CredentialOnTheWay"
      >
        CredentialOffer.CredentialOnTheWay
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "marginTop": 20,
          },
          Object {
            "alignItems": "center",
            "justifyContent": "flex-end",
            "minHeight": 250,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "flexDirection": "column",
            },
          ]
        }
      >
        <
          height={110}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "marginTop": -30,
                "position": "absolute",
              },
            ]
          }
          width={110}
        />
        <View
          style={
            Object {
              "opacity": 1,
              "transform": Array [
                Object {
                  "translateY": -80.9205021307228,
                },
              ],
            }
          }
        >
          <
            height={110}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "marginLeft": 10,
                  "position": "absolute",
                },
              ]
            }
            width={110}
          />
        </View>
        <
          height={140}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
            ]
          }
          width={140}
        />
      </View>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 18,
            "fontWeight": "normal",
          },
          Object {
            "marginTop": 20,
            "textAlign": "center",
          },
        ]
      }
      testID="com.ariesbifold:id/TakingTooLong"
    >
      Connection.TakingTooLong
    </Text>
    <View
      style={
        Array [
          Object {
            "marginBottom": 20,
            "marginTop": "auto",
          },
        ]
      }
    >
      <View
        accessibilityLabel="Loading.BackToHome"
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderColor": "#42803E",
            "borderRadius": 4,
            "borderWidth": 2,
            "opacity": 1,
            "padding": 16,
          }
        }
        testID="com.ariesbifold:id/BackToHome"
      >
        <Text
          style={
            Array [
              Object {
                "color": "#42803E",
                "fontSize": 18,
                "fontWeight": "bold",
                "textAlign": "center",
              },
              false,
            ]
          }
        >
          Loading.BackToHome
        </Text>
      </View>
    </View>
  </RNCSafeAreaView>
</Modal>
`;
